types/types.go

```diff
@@ -494,6 +494,9 @@ type SystemContext struct {
 	// Additional tags when creating or copying a docker-archive.
 	DockerArchiveAdditionalTags []reference.NamedTagged
 
+	// If not "", overrides the temporary directory to use for storing big files
```

Comment:
(Non-blocking nit: Maybe drop the empty line, and the one below `BlobInfoCacheDir`.)